lib/database/user.php

```diff
@@ -1491,54 +1410,34 @@ function GetDeveloperStatsFull($count, $sortBy, $devFilter = 7)
     settype($count, 'integer');
     settype($devFilter, 'integer');
 
-    switch ($devFilter) {
-        case 1: // Active
-            $stateCond = " AND ua.Permissions >= " . Permissions::Developer;
-            break;
-        case 2: // Junior
-            $stateCond = " AND ua.Permissions = " . Permissions::JuniorDeveloper;
-            break;
-        case 3: // Active + Junior
-            $stateCond = " AND ua.Permissions >= " . Permissions::JuniorDeveloper;
-            break;
-        case 4: // Inactive
-            $stateCond = " AND ua.Permissions <= " . Permissions::Registered;
-            break;
-        case 5: // Active + Inactive
-            $stateCond = " AND ua.Permissions <> " . Permissions::JuniorDeveloper;
-            break;
-        case 6: // Junior + Inactive
-            $stateCond = " AND ua.Permissions <= " . Permissions::JuniorDeveloper;
-            break;
-        case 0: // Active + Junior + Inactive
-        case 7:
-        default:
-            $stateCond = "";
-    }
-
-    switch ($sortBy) {
-        case 1: // number of points allocated
-            $order = "ContribYield DESC";
-            break;
-        case 2: // number of achievements won by others
-            $order = "ContribCount DESC";
-            break;
-        case 3:
-            $order = "OpenTickets DESC";
-            break;
-        case 4:
-            $order = "TicketRatio DESC";
-            break;
-        case 5:
-            $order = "LastLogin DESC";
-            break;
-        case 6:
-            $order = "Author ASC";
-            break;
-        case 0:
-        default:
-            $order = "Achievements DESC";
-    }
+    $stateCond = match ($devFilter) {
+        // Active
+        1 => " AND ua.Permissions >= " . Permissions::Developer,
+        // Junior
+        2 => " AND ua.Permissions = " . Permissions::JuniorDeveloper,
+        // Active + Junior
+        3 => " AND ua.Permissions >= " . Permissions::JuniorDeveloper,
+        // Inactive
+        4 => " AND ua.Permissions <= " . Permissions::Registered,
+        // Active + Inactive
+        5 => " AND ua.Permissions <> " . Permissions::JuniorDeveloper,
+        // Junior + Inactive
+        6 => " AND ua.Permissions <= " . Permissions::JuniorDeveloper,
+        // Active + Junior + Inactive
+        default => "",
+    };
+
+    $order = match ($sortBy) {
+        // number of points allocated
+        1 => "ContribYield DESC",
+        // number of achievements won by others
+        2 => "ContribCount DESC",
+        3 => "OpenTickets DESC",
+        4 => "TicketRatio DESC",
+        5 => "LastLogin DESC",
+        6 => "Author ASC",
+        default => "Achievements DESC",
+    };
```

Comment:
These changes will make PHP 8.0 a requirement. (see my question about the upgrade being required in#935). It seems like that needs to happen first, but the title of this PR suggests it's a prerequisite for PHP 8. I guess they both have to go together? 

Also, is this still a draft?

Reply:
> the title of this PR suggests it's a prerequisite for PHP 8

Oh, that's misleading then, sorry.

As mentioned in the other PR, this PR here will use PHP 8 language features and require the application to actually run on PHP 8 whereas https://github.com/RetroAchievements/RAWeb/pull/935 merely takes care of a few deprecation warnings that surface when running PHP 7.4 code with PHP 8.

Now that https://github.com/RetroAchievements/RAWeb/pull/935 is merged we can safely switch to PHP 8 on the server.

After that's done this PR can be merged and deployed.
I'm still finding things to clean up, that's why I haven't yet removed the draft label. Will be ready very soon.

Reply:
PR is ready now - I have to stop looking for things to clean up, it's getting out of hand.